src/instructlab/data/generate.py

```diff
@@ -210,7 +210,9 @@ def generate(
         from instructlab.model.backends import backends
         from instructlab.model.backends.llama_cpp import Server as llama_cpp_server
 
+        # TODO (cdoern): we really should not edit the cfg object
```

Comment:
ok then why not just pass the `model` into `select_backend`?

Reply:
`select_backend` should be considered relatively frozen ATM. I don't really want to alter functionality those close to a release which many components depend on

Reply:
I mean... you pass the value into the function as an argument, no? see line 216. why not just pass it without setting on `ctx.obj`?

Reply:
Disregard, I misread that `select_backend` doesn't accept just the model, but the whole config section.

Reply:
But I think you are breaking customizing the model via config file:


```
$ cat ~/.config/instructlab/config.yaml | grep -A 8 teacher
  teacher:
    backend: null
    chat_template: null
    host_port: 127.0.0.1:8000
    llama_cpp:
      gpu_layers: -1
      llm_family: ''
      max_ctx_size: 4096
    model_path: INVALID_MODEL
$ .tox/py311-unit/bin/ilab data generate
INFO 2024-07-29 15:58:21,301 numexpr.utils:161: NumExpr defaulting to 4 threads.
INFO 2024-07-29 15:58:21,741 datasets:58: PyTorch version 2.3.1+cpu available.
Model used: /home/vagrant/.cache/instructlab/models/merlinite-7b-lab-Q4_K_M.gguf

```

Reply:
(that is with:)

```
diff --git a/src/instructlab/data/generate.py b/src/instructlab/data/generate.py
index 1da2aba..29b35ec 100644
--- a/src/instructlab/data/generate.py
+++ b/src/instructlab/data/generate.py
@@ -213,6 +213,7 @@ def generate(
         # TODO (cdoern): we really should not edit the cfg object
         ctx.obj.config.serve.llama_cpp.llm_family = model_family
         ctx.obj.config.generate.teacher.model_path = model
+        print("Model used:", ctx.obj.config.generate.teacher.model_path)
         backend_instance = backends.select_backend(ctx.obj.config.generate.teacher)

         try:

```